core/property/registry.go

```diff
@@ -2,31 +2,68 @@ package property
 
 import (
 	"fmt"
-	"sync"
-
+	"github.com/TIBCOSoftware/flogo-lib/config"
 	"github.com/TIBCOSoftware/flogo-lib/core/data"
 	"github.com/TIBCOSoftware/flogo-lib/logger"
+	"reflect"
+	"regexp"
+	"sync"
 )
 
 var (
-	props = make(map[string]string)
-	mut   = sync.RWMutex{}
+	props    = make(map[string]interface{})
+	mut      = sync.RWMutex{}
+	regex    = regexp.MustCompile(config.GetPropertyDelimiterFormat())
+	resolver Resolver
 )
 
+func init() {
```

Comment:
This init method should move to the EnvVarResolver we define.   the registry should be done by resolver itself.